pkg/nsx/services/builder.go

```diff
@@ -639,7 +689,38 @@ func (service *SecurityPolicyService) buildRuleDstGroup(obj *v1alpha1.SecurityPo
 	return &ruleDstGroup, ruleDstGroupPath, err
 }
 
-func (service *SecurityPolicyService) buildPeerTags(obj *v1alpha1.SecurityPolicy, peers *[]v1alpha1.SecurityPolicyPeer, idx int) []model.Tag {
+// Build rule basic info, ruleIdx is the index of the rules of security policy,
+// portIdx is the index of rule's ports, portAddressIdx is the index
+// of multiple port number if one named port maps to multiple port numbers.
+func (service *SecurityPolicyService) buildRuleBasicInfo(obj *v1alpha1.SecurityPolicy, rule *v1alpha1.SecurityPolicyRule, ruleIdx int, portIdx int, portAddressIdx int) (*model.Rule, error) {
+	nsxRuleID := service.buildRuleID(obj, ruleIdx)
+	nsxRuleName := service.buildRuleName(obj, rule, ruleIdx)
+	ruleAction, err := getRuleAction(rule)
+	if err != nil {
+		return nil, err
+	}
+	ruleDirection, err := getRuleDirection(rule)
+	if err != nil {
+		return nil, err
+	}
+	sequence := int64(ruleIdx)
+	rID := fmt.Sprintf("%s_%d_%d", nsxRuleID, portIdx, portAddressIdx)
+	rName := fmt.Sprintf("%s-%d-%d", nsxRuleName, portIdx, portAddressIdx)
+
+	nsxRule := model.Rule{
+		Id:             &rID,
+		DisplayName:    &rName,
+		Direction:      &ruleDirection,
+		SequenceNumber: &sequence,
+		Action:         &ruleAction,
+		Services:       []string{"ANY"},
+		Tags:           service.buildBasicTags(obj),
+	}
+	log.V(1).Info("built rule basic info", "nsxRule", nsxRule)
+	return &nsxRule, nil
+}
+
+func (service *SecurityPolicyService) BuildPeerTags(obj *v1alpha1.SecurityPolicy, peers *[]v1alpha1.SecurityPolicyPeer, idx int) []model.Tag {
```

Comment:
Because here,
				patches := gomonkey.ApplyMethod(reflect.TypeOf(s), "BuildPeerTags",
		func(s *SecurityPolicyService, v *v1alpha1.SecurityPolicy, p *[]v1alpha1.SecurityPolicyPeer, i int) []model.Tag {
			peerTags := []model.Tag{
				{Scope: nil, Tag: nil},
			}
			return peerTags
		})
I have to change it to upper case, otherwise, it can't gomonkey it.